### PR TITLE
General cleanup of webrtc/basics/en/index.html

### PR DESCRIPTION
**Pull request by KaptenJansson**
* Fixed broken links
* Updated version information
* Add Microsoft edge as mediastream capable
* Update links to point to the WebRTC/samples repo in favor of simpl
* Fixed indentation to two spaces
* Add notes about getUserMedia is a power feature
* Add bug trackers
* Add link to webrtc/samples repo in quick start
* Add link to test.webrtc.org in quick start

### CODE REVIEW

**Comment by KaptenJansson** [comment]
@samdutton PTAL

**Comment by KaptenJansson** [comment]
There are more things I want to change but I think we need to split it up a little.

File: content/tutorials/webrtc/basics/en/index.html [hunk 1]
```diff
@@ -50,7 +50,9 @@ <h2 id="toc-tldr">Quick start</h2>
...
   <li>Get to grips with the RTCPeerConnection API by reading through the <a href="#simpleRTCPeerConnectionExample" title="Internal link to simple RTCPeerConnecton example">simple example below</a> and the demo at <a href="http://www.simpl.info/pc" title="WebRTC demo without signaling">simpl.info/pc</a>, which implements WebRTC on a single web page.</li>
-	<li>Learn more about how WebRTC uses servers for signaling, and firewall and NAT traversal, by reading through the code and console logs from <a href="http://apprtc.appspot.com" title="Simple WebRTC video chat demo">apprtc.appspot.com</a>.</li>
+  <li>Learn more about how WebRTC uses servers for signaling, and firewall and NAT traversal, by reading through the code and console logs from <a href="https://apprtc.appspot.com" title="Simple WebRTC video chat demo">apprtc.appspot.com</a>.</li>
+  <li>Can’t wait and just want to try out WebRTC right now? Try out some of the <a href="https://webrtc.github.io/samples" title="WebRTC Samples">20+ demos</a> that exercise the WebRTC JavaScript APIs</li>
```
Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: +1 to these additions.

File: content/tutorials/webrtc/basics/en/index.html [hunk 2]
```diff
@@ -74,6 +76,9 @@ <h2 id="toc-history">A very short history of WebRTC</h2>
 <p>The guiding principles of the WebRTC project are that its APIs should be open source, free, standardized, built into web browsers and more efficient than existing technologies.</p>
 
 <h2 id="toc-where">Where are we now?</h2>
+<p>WebRTC is used in various apps like Whatsapp, FaceTime messenger, appear.in and platforms such as TokBox. There is even an experimental WebRTC enabled iOS Browser named Bowser. WebRTC has also been integrated with <a href="https://labs.ericsson.com/developer-community/blog/beyond-html5-conversational-voice-and-video-implemented-webkit-gtk" title="Ericsson article about WebKitGTK+">WebKitGTK+</a> and <a href="http://www.youtube.com/watch?v=Vm5ebKWKNE8" title="basysKom showcasing WebRTC based video chat in QML application">Qt</a> native apps.</p>
```
Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: @juberti: OK to mention these?

Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: WhatsApp

File: content/tutorials/webrtc/basics/en/index.html [hunk 3]
```diff
@@ -74,6 +76,9 @@ <h2 id="toc-history">A very short history of WebRTC</h2>
...
+
+<p>Also Microsoft <a href="http://blogs.windows.com/msedgedev/2015/05/13/announcing-media-capture-functionality-in-microsoft-edge/" title="media capture functionality in Microsoft Edge">recently</a> added MediaCapture and Stream APIs to Edge.</p>
```
Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: Could you rephrase this so as not to say 'recently', maybe just 'Microsoft has added MediaCapture and Stream APIs to Edge.'

File: content/tutorials/webrtc/basics/en/index.html [hunk 4]
```diff
@@ -82,13 +87,12 @@ <h2 id="toc-where">Where are we now?</h2>
...
...de><strong>RTCPeerConnection</strong></code> is in Chrome (on desktop and for Android), Opera (on desktop and in the latest Android Beta) and in Firefox. A word of explanation about the name: after several iterations, <code>RTCPeerConnection</code> is currently implemented by Chrome and Opera as <code>webkitRTCPeerConnection</code> and by Firefox as <code>mozRTCPeerConnection</code>. Other names and implementations have been deprecated. When the standards process has stabilized, the prefixes will be removed. There's an ultra-simple demo of Chromium's RTCPeerConnection implementation at <a href="https://webrtc.github.io/samples/src/content/peerconnection/pc1/" title="Simple cross-platform peerconnection demo">GitHub</a> and a great video chat application at <a href="https://apprtc.appspot.com" title="Video chat demo">apprtc.appspot.com</a>. This app uses <a href="https://github.com/webrtc/adapter" title="adapter.js JavaScript file">adapter.js</a>, a JavaScript shim, maintained Google with help from the <a href="https://github.com/webrtc/adapter/graphs/contributors" title="WebRTC/adapter contributors">GitHub community</a>, that abstracts away browser differences and spec changes.</p>
```
Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: Probably 'WebRTC community' rather than 'GitHub community'.

File: content/tutorials/webrtc/basics/en/index.html [hunk 5]
```diff
@@ -82,13 +87,12 @@ <h2 id="toc-where">Where are we now?</h2>
...
 
-<p>WebRTC functionality is available in Internet Explorer <a href="https://groups.google.com/forum/#!topic/discuss-webrtc/tKoh1wrI8ig" title="How to enable WebRTC functionality in Internet Explorer via Chrome Frame">via Chrome Frame</a>, and Skype (acquired by Microsoft in 2011) is reputedly <a href="http://gigaom.com/2012/06/26/skype-webrtc-web-client/" title="GigaOM blog post about Skype and WebRTC">planning to use WebRTC</a>. WebRTC has also been integrated with <a href="https://labs.ericsson.com/developer-community/blog/beyond-html5-conversational-voice-and-video-implemented-webkit-gtk" title="Ericsson article about WebKitGTK+">WebKitGTK+</a> and <a href="http://www.youtube.com/watch?v=Vm5ebKWKNE8" title="basysKom showcasing WebRTC based video chat in QML application">Qt</a> native apps.</p>
+<p><strong><code>RTCDataChannel</code></strong> is supported by Chrome M25, Opera 18 and Firefox 22 and above. Check out one of the data channel demos at <a href="https://webrtc.github.io/samples/" title="WebRTC samples">GitHub</a> to see it in action.
```
Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: I think you could just say 'supported by Chrome, Opera and Firefox'. The rest is ancient history :).

File: content/tutorials/webrtc/basics/en/index.html [hunk 6]
```diff
@@ -122,7 +126,7 @@ <h2 id="toc-mediastream">MediaStream (aka getUserMedia)</h2>
 <p>Probably the easiest way to understand MediaStream is to look at it in the wild:</p>
 
 <ol>
-  <li>In Chrome or Opera, open the demo at <a href="http://simpl.info/getusermedia/" title="simpl.info getUserMedia demo">simpl.info/gum</a>.</li>
+  <li>In Chrome or Opera, open the demo at <a href="https://webrtc.github.io/samples/src/content/getusermedia/gum/" title="getUserMedia demo">https://webrtc.github.io/samples/src/content/getusermedia/gum</a>.</li>
```
Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: +1 to changing these.

File: content/tutorials/webrtc/basics/en/index.html [hunk 7]
```diff
@@ -165,6 +169,8 @@ <h2 id="toc-mediastream">MediaStream (aka getUserMedia)</h2>
 
 <p>Likewise on pages using HTTPS: permission only has to be granted once for for <code>getUserMedia()</code> (in Chrome at least). First time around, an Always Allow button is displayed in the browser's <a href="http://dev.chromium.org/user-experience/infobars" title="Chromium information about the browser infobar">infobar</a>.</p>
 
+<p>Note that HTTP access for getUserMedia() will be deprecated soon in Chrome due to it being classified as a <a href="https://sites.google.com/a/chromium.org/dev/Home/chromium-security/deprecating-powerful-features-on-insecure-origins" title="Chrome powerful features">Powerful feature</a>. You can already see a warning when invoked on a HTTP page on Chrome M44.</p>
```
Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: Could you rephrase this so it will sound OK in the future.

Say 'Note that HTTP access for <code>getUserMedia()</code> is deprecated in Chrome from version XX. ...'

Also, Chrome

File: content/tutorials/webrtc/basics/en/index.html [hunk 8]
```diff
@@ -184,9 +190,9 @@ <h2 id="toc-mediastream">MediaStream (aka getUserMedia)</h2>
...
+<p><a href="http://tools.ietf.org/html/draft-alvestrand-constraints-resolution-00#page-4" title="IETF Resolution Constraints draft specification">Constraints</a> have been implemented since Chrome M24, Firefox 32 and Opera 18. These can be used to set values for video resolution for <code>getUserMedia()</code> and RTCPeerConnection <code>addStream()</code> calls. The intention is to implement <a href="http://dev.w3.org/2011/webrtc/editor/getusermedia.html#the-model-sources-sinks-constraints-and-states" title="W3C getUserMedia Editor's Draft - The model: sources, sinks, constraints, and states">support for other constraints</a> such as aspect ratio, facing mode (front or back camera), frame rate, height and width, along with an <a href="http://dev.w3.org/2011/webrtc/editor/getusermedia.html#methods-1" title="applyConstraints() API proposal in W3C getUserMedia Editor's Draft"><code>applyConstraints()</code> method</a>. </p>
```
Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: I think we could get rid of versions: 'Constraints are implemented in Chrome, Firefox and Opera.'

File: content/tutorials/webrtc/basics/en/index.html [hunk 9]
```diff
@@ -197,15 +203,15 @@ <h3 id="toc-constraints">Constraints</h3>
...
 
-<p>Chrome apps also now make it possible to share a live 'video' of a single browser tab via the experimental <a href="http://developer.chrome.com/dev/extensions/tabCapture.html" title="chrome.tabCapture API documentation">chrome.tabCapture</a> API. For a screencast, code and more information, see the HTML5 Rocks update: <a href="http://updates.html5rocks.com/2012/12/Screensharing-with-WebRTC" title="HTML5 Rocks update article: Screensharing with WebRTC">Screensharing with WebRTC</a>. This feature is not yet available in Opera.</p>
+<p>Chrome apps also now make it possible to share a live 'video' of a single browser tab via the experimental <a href="http://developer.chrome.com/dev/extensions/tabCapture.html" title="chrome.tabCapture API documentation">chrome.tabCapture</a> API. For screencast, code and more information, see the HTML5 Rocks update: <a href="http://updates.html5rocks.com/2012/12/Screensharing-with-WebRTC" title="HTML5 Rocks update article: Screensharing with WebRTC">Screensharing with WebRTC</a>. This feature is not yet available in Opera.</p>
```
Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: This could be deleted, right?: 'This feature is not yet available in Opera.'

File: content/tutorials/webrtc/basics/en/index.html [hunk 10]
```diff
@@ -435,16 +441,16 @@ <h3 id="toc-real">RTCPeerConnection plus servers</h3>
...
+<p>A good place to try out WebRTC, complete with signaling and NAT/firewall traversal using a STUN server, is the video chat demo at <a href="https://apprtc.appspot.com" title="Simple WebRTC demo" target="_blank">apprtc.appspot.com</a>. This app uses <a href="https://github.com/webrtc/adapter" title="adapter.js JavaScript file">adapter.js</a> to cope with different RTCPeerConnection and <code>getUserMedia()</code> implementations.</p>
 
 <p>The code is deliberately verbose in its logging: check the console to understand the order of events. Below we give a detailed walk-through of the code.</p>
 
 <h3>What's going on?</h3>
 
-<p>The demo starts by running the <code>initalize()</code> function:</p>
+<p>The demo starts by running the <code>initialize()</code> function:</p>
```
Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: Great spot!

File: content/tutorials/webrtc/basics/en/index.html [hunk 11]
```diff
@@ -777,17 +783,33 @@ <h2 id="toc-conclusion">In conclusion</h2>
...
+        </figure>
+      </li>
+    </ul>
   </li>
-  <li>Cross browser <a href="http://www.webrtc.org/interop" title="webrtc.org Firefox/Chrome interop information">interop notes</a></li>
-  <li><a href="https://apprtc.appspot.com/js/adapter.js" title="adapter.js JavaScript file">adapter.js</a> is a JavaScript shim for WebRTC, maintained by Google, that abstracts vendor prefixes, browser differences and spec changes</li>
-  <li>To learn more about WebRTC signaling processes, check the <a href="http://apprtc.appspot.com" title="apprtc.appspot.com video chat demo">apprtc.appspot.com</a> log output to the console</li>
+  <li>Cross browser <a href="http://www.webrtc.org/web-apis/interop" title="webrtc.org Firefox/Chrome interop information">interop notes</a></li>
+  <li><a href="https://github.com/webrtc/adapter" title="adapter.js JavaScript file">adapter.js</a> is a JavaScript shim for WebRTC, maintained by Google with help from the <a href="https://github.com/webrtc/adapter/graphs/contributors" title="WebRTC/adapter contributors">GitHub community</a>, that abstracts vendor prefixes, browser differences and spec changes</li>
```
Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: Maybe 'WebRTC community' as above.

File: content/tutorials/webrtc/basics/en/index.html [hunk 12]
```diff
@@ -777,17 +783,33 @@ <h2 id="toc-conclusion">In conclusion</h2>
...
+  <li>To learn more about WebRTC signaling processes, check the <a href="https://apprtc.appspot.com" title="apprtc.appspot.com video chat demo">apprtc.appspot.com</a> log output to the console</li>
   <li>If it's all too much, you may prefer to use a <a href="http://io13webrtc.appspot.com/#69" title="WebRTC frameworks">WebRTC framework</a> or even a complete <a href="http://io13webrtc.appspot.com/#72" title="WebRTC service providers">WebRTC&nbsp;service</a></li>
-  <li>Bug reports and feature requests are always appreciated: <a href="http://www.crbug.com/new" title="Report Chrome bugs and feature requests">crbug.com/new</a> for Chrome bugs, <a href="https://bugs.opera.com/wizard/" title="Report Opera bugs and feature requests">bugs.opera.com/wizard/</a> for Opera, <a href="https://bugzilla.mozilla.org/" title="File a Firefox bug">bugzilla.mozilla.org</a> for Firefox</li>
+  <li>Bug reports and feature requests are always appreciated:
+    <ul>
+      <li><a href="https://code.google.com/p/webrtc/issues/entry" title="Report WebRTC bugs and feature requests">bugs.webrtc.org/new</a> for WebRTC bugs</li>
```
Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: Now this a list, I think we could get rid of the 'for '.

File: content/tutorials/webrtc/basics/en/index.html [hunk 13]
```diff
@@ -797,11 +819,12 @@ <h2 id="toc-more">Learn more</h2>
...
   <li><a href="http://www.webrtc.org/" title="webrtc.org">webrtc.org</a> is home to all things WebRTC: demos, documentation and discussion</li>
-  <li><a href="http://www.webrtc.org/demo" title="webrtc.org demos">webrtc.org demo page</a>: links to demos</li>
+  <li><a href="https://github.com/webrtc/samples" title="webrtc.org demos">webrtc.org demo page</a>: links to demos</li>
   <li><a href="https://groups.google.com/forum/?fromgroups#!forum/discuss-webrtc" title="discuss-webrt Google Group">discuss-webrtc</a>: Google Group for technical WebRTC discussion</li>
   <li><a href="https://plus.sandbox.google.com/113817074606039822053/posts" title="WebRTC on Google+">+webrtc</a></li>
   <li><a href="https://twitter.com/webrtc" title="WebRTC on Twitter">@webrtc</a></li>
   <li>Google Developers <a href="https://developers.google.com/talk/libjingle/important_concepts#connections" title="Google Developers: Google Talk for Developers">Google Talk documentation</a>, which gives more information about NAT traversal, STUN, relay servers and candidate gathering</li>
+  <li><a href="https://github.com/webrtc" title="WebRTC on GitHub">WebRTC on github</a></li>
```
Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: GitHub

**Comment by samdutton** [comment]
Great job! LGTM apart from inline nitpicks. 

I'll also get a tech writer to give the article a once over once these changes are pushed.

File: content/tutorials/webrtc/basics/en/index.html (hunk 7 above)
Comment (KaptenJansson) [inline content/tutorials/webrtc/basics/en/index.html]: I do not know which versions it actually will be deprecated, currently the powerful feature article states 2015.  Will add end of 2015 instead for now.

File: content/tutorials/webrtc/basics/en/index.html (hunk 5 above)
Comment (KaptenJansson) [inline content/tutorials/webrtc/basics/en/index.html]: Done

File: content/tutorials/webrtc/basics/en/index.html (hunk 4 above)
Comment (KaptenJansson) [inline content/tutorials/webrtc/basics/en/index.html]: Done

File: content/tutorials/webrtc/basics/en/index.html (hunk 3 above)
Comment (KaptenJansson) [inline content/tutorials/webrtc/basics/en/index.html]: Done

File: content/tutorials/webrtc/basics/en/index.html (hunk 2 above)
Comment (KaptenJansson) [inline content/tutorials/webrtc/basics/en/index.html]: Done

File: content/tutorials/webrtc/basics/en/index.html (hunk 1 above)
Comment (KaptenJansson) [inline content/tutorials/webrtc/basics/en/index.html]: Done

File: content/tutorials/webrtc/basics/en/index.html (hunk 9 above)
Comment (KaptenJansson) [inline content/tutorials/webrtc/basics/en/index.html]: I just tried, does not seem like Opera supports screensharing yet.

File: content/tutorials/webrtc/basics/en/index.html (hunk 11 above)
Comment (KaptenJansson) [inline content/tutorials/webrtc/basics/en/index.html]: Done

File: content/tutorials/webrtc/basics/en/index.html (hunk 12 above)
Comment (KaptenJansson) [inline content/tutorials/webrtc/basics/en/index.html]: Done

File: content/tutorials/webrtc/basics/en/index.html (hunk 13 above)
Comment (KaptenJansson) [inline content/tutorials/webrtc/basics/en/index.html]: Done

**Comment by KaptenJansson** [comment]
Also in a separate PR we might want to go over the code examples so they match the linked demo pages.

File: content/tutorials/webrtc/basics/en/index.html (hunk 8 above)
Comment (KaptenJansson) [inline content/tutorials/webrtc/basics/en/index.html]: Done

File: content/tutorials/webrtc/basics/en/index.html (hunk 9 above)
Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: I meant that it maybe wasn't worth pointing out that screensharing isn't supported by Opera.

Maybe rephrase: 'It's also possible to use screen capture as a MediaStream source in Chrome using the experimental ...'

**Comment by samdutton** [comment]
Good idea to check code examples.

LGTM. If not objections @PaulKinlan, I'll merge.

**Comment by KaptenJansson** [comment]
Ping @samdutton and @PaulKinlan

**Comment by samdutton** [comment]
Ping @PaulKinlan :).

**Comment by samdutton** [comment]
👍

**Comment by samdutton** [comment]
Any idea when these changes will be published?

File: content/tutorials/webrtc/basics/en/index.html [hunk 14]
```diff
@@ -74,6 +76,9 @@ <h2 id="toc-history">A very short history of WebRTC</h2>
 <p>The guiding principles of the WebRTC project are that its APIs should be open source, free, standardized, built into web browsers and more efficient than existing technologies.</p>
 
 <h2 id="toc-where">Where are we now?</h2>
+<p>WebRTC is used in various apps like WhatsApp, FaceTime messenger, appear.in and platforms such as TokBox. There is even an experimental WebRTC enabled iOS Browser named Bowser. WebRTC has also been integrated with <a href="https://labs.ericsson.com/developer-community/blog/beyond-html5-conversational-voice-and-video-implemented-webkit-gtk" title="Ericsson article about WebKitGTK+">WebKitGTK+</a> and <a href="http://www.youtube.com/watch?v=Vm5ebKWKNE8" title="basysKom showcasing WebRTC based video chat in QML application">Qt</a> native apps.</p>
```
Comment (juberti) [inline content/tutorials/webrtc/basics/en/index.html]: This should say "Facebook Messenger", not FaceTime messenger

Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: Fixed in https://github.com/html5rocks/www.html5rocks.com/pull/1398.

Comment (samdutton) [inline content/tutorials/webrtc/basics/en/index.html]: @PaulKinlan @petele Before publishing we should make sure to merge #1398.

**Comment by samdutton** [comment]
@PaulKinlan @petele Could we get these fixes published? This article gets about a thousand views per day.
